cmd/kubeadm/app/phases/etcd/local.go

```diff
@@ -134,32 +134,45 @@ func RemoveStackedEtcdMemberFromCluster(client clientset.Interface, cfg *kubeadm
 // CreateStackedEtcdStaticPodManifestFile will write local etcd static pod manifest file
 // for an additional etcd member that is joining an existing local/stacked etcd cluster.
 // Other members of the etcd cluster will be notified of the joining node in beforehand as well.
-func CreateStackedEtcdStaticPodManifestFile(client clientset.Interface, manifestDir, patchesDir string, nodeName string, cfg *kubeadmapi.ClusterConfiguration, endpoint *kubeadmapi.APIEndpoint, isDryRun bool) error {
+func CreateStackedEtcdStaticPodManifestFile(client clientset.Interface, manifestDir, patchesDir string, nodeName string, cfg *kubeadmapi.ClusterConfiguration, endpoint *kubeadmapi.APIEndpoint, isDryRun bool, certificatesDir string) error {
 	// creates an etcd client that connects to all the local/stacked etcd members
 	klog.V(1).Info("creating etcd client that connects to etcd pods")
-	etcdClient, err := etcdutil.NewFromCluster(client, cfg.CertificatesDir)
+	etcdClient, err := etcdutil.NewFromCluster(client, certificatesDir)
 	if err != nil {
 		return err
 	}
 
 	etcdPeerAddress := etcdutil.GetPeerURL(endpoint)
 
-	klog.V(1).Infof("[etcd] Adding etcd member: %s", etcdPeerAddress)
-	var cluster []etcdutil.Member
-	cluster, err = etcdClient.AddMember(nodeName, etcdPeerAddress)
+	klog.V(1).Infoln("[etcd] Getting the list of existing members")
+	cluster, err := etcdClient.ListMembers()
```

Comment:
the logic in this function changed in a recent PR.
we should remove the ListMembers here.

https://github.com/kubernetes/kubernetes/blob/master/cmd/kubeadm/app/phases/etcd/local.go#L137-L169

we should:
- add member (AddMember) and wait for cluster (WaitForClusterAvailable) only if not dry-running.
- save file to temp folder for dry-run and normal folder if not.

Reply:
@neolit123 
Yeah,I noticed that and preserved ListMembers here because the init cluster info used in prepareAndWriteEtcdStaticPod should be available when dryrun. 
AddMember can't return existing member list for dryrun without pass a existed member, is there a better way to do?

Reply:
we should not call ListMembers before hand, as a recent PR tried to fix related (rare) problems.
during dry-run you could pass `cluster` to be an empty `[]etcdutil.Member`.

Reply:
Thanks, got it!
remove ListMembers now